src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,194 @@
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car: " + car + ". ", e);
+        }
+        removeDriversFromCar(car);
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car with id: " + id + ". ", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * FROM cars_drivers cd JOIN cars c ON cd.car_id = c.id "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars by driver with id: "
+                    + driverId + ". ", e);
+        }
+
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("id", Long.class));
+        car.setModel(resultSet.getString("model"));
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject("manufacturer_id", Long.class));
+        manufacturer.setName(resultSet.getString("name"));
+        manufacturer.setCountry(resultSet.getString("country"));
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private Driver getDriver(ResultSet resultSet) throws SQLException {
+        Driver driver = new Driver();
+        driver.setId(resultSet.getObject("id", Long.class));
+        driver.setName(resultSet.getString("name"));
+        driver.setLicenseNumber(resultSet.getString("license_number"));
+        return driver;
+    }
+
+    private List<Driver> getDriversFromCar(Car car) {
+        String query = "SELECT id, name, license_number FROM cars_drivers cd JOIN drivers d "
+                + "ON cd.driver_id = d.id WHERE d.is_deleted = false AND cd.car_id = ?;";
```

Comment:
```suggestion
        String query = "SELECT d.id, d.name, d.license_number FROM cars_drivers cd JOIN drivers d "
                + "ON cd.driver_id = d.id WHERE d.is_deleted = false AND cd.car_id = ?;";
```
if you write alias for table, use it with columns